sdks/python/apache_beam/runners/worker/opcounters.py

```diff
@@ -224,8 +230,25 @@ def _observable_callback_inner(value, is_encoded=False):
 
     return _observable_callback_inner
 
+  def type_check(self, value):
+    # type: (any, bool) -> None
+    for transform_label, type_constraint_tuple in self.output_type_constraints.items():
+      parameter_name, constraint = type_constraint_tuple
+      if constraint is not None:
```

Comment:
TODO: During graph construction, don't add any `None` constraints. This will allow us to remove this check

Reply:
Yes, please do that.